### PR TITLE
ci: run integration test in GitHub actions

### PR DESCRIPTION
**Pull request by merll**
This PR implements redkubes/unassigned-issues#418.

The following might want to be changed, but is possibly out of scope of this issue:
- [ ] Probably we also want to run tests on a feature branch before merging into main. We have to discuss a way to do this on-demand (e.g. contents of a commit message) since it takes a lot of time to run this each time.
- [ ] This is not new, but currently all images with successful unit-tests are pushed to Docker Hub. That might not be intended. We probably want to have this only done for `main` and release tags.

## Checklist

- [ ] Architecture Design Records have been added as `adr/*.md` and appended to list in `adr/_index.md`, if applicable.
- [x] The `values-schema.yaml` file and `test/**` fixtures have been updated to reflect code changes, if applicable.
- [ ] The OpenApi Schema from redkubes/otomi-api project is compatible with definitions from `values-schema.yaml` file, if applicable.
- [ ] Helm releases are meeting otomi's baseline security policies, if applicable.
- [ ] Helm chart and helmfile changes are tested against upgrade scenario, if applicable.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ env:
   CACHE_REPO: redkubes/otomi-core
   REPO: otomi/core
   GIT_USER: redkubesbot
+  DIGITALOCEAN_PROJECT: 888d7eee-cd33-4f22-8499-274c393f9b80
```
Comment (diabhey) [inline .github/workflows/main.yml:13]: Suggestion to remove hardcoded values and use secrets instead.

Comment (merll) [reply .github/workflows/main.yml]: Done.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -51,46 +52,84 @@ jobs:
...
+      - name: Install doctl
+        uses: digitalocean/action-doctl@v2
+        with:
+          token: ${{ secrets.DIGITALOCEAN_ACCESS_TOKEN }}
+      - name: Create k8s cluster for testing
+        run: |
+          doctl kubernetes cluster create integration-test-${{ github.sha }} \
+            --tag integration-test \
+            --region ams3 \
+            --vpc-uuid=2c840be3-de75-48c4-b0d6-66b214e2d39e \
```
[the remark this answers is not in the document]
Comment (merll) [reply .github/workflows/main.yml]: `vpc-uuid` is now retrieved automatically, and `region` is an input value with default `ams3`.

**Comment by Morriz** [comment]
Any reason to create and then destroy a slow installing cluster for tests? I really favor having one dev cluster always running and then using vcluster as @diabhey suggested. That way we can run multiple tests side by side very quickly.

**Comment by Morriz** [comment]
And is there any reason to have a minimal install when a full install covers everything? I'd drop the minimal install as it gives no extra assurances imo. Please give arguments if I'm mistaken here.

As discussed in a bigger story, which aims at reaching better configuration coverage, we should focus on strategies that allow superfast deployment and design an algorithm that generates a list (with a max length) of value sets, that touches >60% in one run, and will reach 100% coverage over multiple runs. So not every run has to reach 100%, as that would require way too many permutations.

Unfortunately it is

**Comment by j-zimnowoda** [comment]
> Any reason to create and then destroy a slow installing cluster for tests? I really favor having one dev cluster always running and then using vcluster as @diabhey suggested. That way we can run multiple tests side by side very quickly.

The cluster creation is 5m50s and is performed in parallel with build-test-cache. It means that there will be just 2-3m more waiting time. BUT we can create many clusters in parallel and perform tests in parallel.
 
The full installation is 25m29s. If we really want to gain time during integration tests then we should rather focus on improving installation time.

**Comment by diabhey** [comment]
https://github.com/hivenetes/hive/blob/main/.github/workflows/otomi-vcluster.yaml - Here is a sample POC, in case we want to take the vCluster approach.

**Comment by j-zimnowoda** [comment]
Looking good. 
Regarding your TODO items:
1) yes, I believe that on feature branch you should be able to trigger tests manually 
2) indeed all builds are pushed to dockerhub and it looks a bit messy right now. It could be changed but you need to remember that on a daily basis developers deploy otomi from feature branches by just changing `otomi.versions` parameter.  If registry is changed then there has to be an option to provide it via values files.

**Comment by j-zimnowoda** [comment]
> And is there any reason to have a minimal install when a full install covers everything? I'd drop the minimal install as it gives no extra assurances imo. Please give arguments if I'm mistaken here.
> 
> As discussed in a bigger story, which aims at reaching better configuration coverage, we should focus on strategies that allow superfast deployment and design an algorithm that generates a list (with a max length) of value sets, that touches >60% in one run, and will reach 100% coverage over multiple runs. So not every run has to reach 100%, as that would require way too many permutations.
> 
> Unfortunately it is

A developer may want to trigger manually minimal install. Let's see if it is used. If not then we will remove it.

**Review by j-zimnowoda — APPROVED** [review]

**Comment by j-zimnowoda** [comment]
> https://github.com/hivenetes/hive/blob/main/.github/workflows/otomi-vcluster.yaml - Here is a sample POC, in case we want to take the vCluster approach.

That is really cool. So there must be just one long running cluster that scales down. I propose to incorporate vcluster in the next step.

**Pull request merged**